src/mixins/filing-template-mixin.ts

```diff
@@ -334,18 +326,180 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
   }
 
   /**
-   * Builds an incorporation application filing from store data. Used when saving a filing.
+   * Builds an continuation in filing from store data. Used when saving a filing.
    * @returns the filing body to save
    */
-  buildIncorporationFiling (): IncorporationFilingIF {
-    function fixNullAddressType (orgPeople: OrgPersonIF[]): OrgPersonIF[] {
-      return orgPeople.map(p => {
-        if (p.deliveryAddress?.addressType === null) delete p.deliveryAddress.addressType
-        if (p.mailingAddress?.addressType === null) delete p.mailingAddress.addressType
-        return p
+  buildContinuationInFiling (): ContinuationInFilingIF {
+    // Build the main filing.
+    const filing: ContinuationInFilingIF = {
+      header: {
+        name: FilingTypes.CONTINUATION_IN,
+        certifiedBy: this.getCertifyState.certifiedBy,
+        date: this.getCurrentDate,
+        filingId: this.getFilingId,
+        folioNumber: this.getFolioNumber,
+        isFutureEffective: this.getEffectiveDateTime.isFutureEffective
+      },
+      business: {
+        legalType: this.getEntityType,
+        identifier: this.getTempId
+      },
+      continuationIn: {
+        foreignJurisdiction: '',
+        nameRequest: {
+          legalType: this.getEntityType
+        },
+        nameTranslations: this.getNameTranslations,
+        offices: this.getDefineCompanyStep.officeAddresses,
+        contactPoint: {
+          email: this.getBusinessContact.email,
+          phone: this.getBusinessContact.phone,
+          // don't save extension if it's empty
+          ...this.getBusinessContact.extension
+            ? { extension: +this.getBusinessContact.extension }
+            : {}
+        },
+        parties: this.fixNullAddressType(this.getAddPeopleAndRoleStep.orgPeople),
+        shareStructure: {
+          shareClasses: this.getCreateShareStructureStep.shareClasses
+        }
+      }
+    }
+
+    // Add business name data.
+    switch (this.getCorrectNameOption) {
+      case CorrectNameOptions.CORRECT_AML_ADOPT:
```

Comment:
I don't know if you'll have these 3 options for cont in filings, but you can clean this up later.

Reply:
Yes, there's a ticket to add the component and deal/clean these fields.